bundles/edu.kit.ipd.sdq.activextendannotations/src/edu/kit/ipd/sdq/activextendannotations/DelegateProcessorUtil.xtend

```diff
@@ -1,65 +0,0 @@
-package edu.kit.ipd.sdq.activextendannotations
-
-import org.eclipse.xtend.lib.macro.TransformationContext
-import org.eclipse.xtend.lib.macro.declaration.MutableClassDeclaration
-import org.eclipse.xtend.lib.macro.declaration.ResolvedMethod
-import org.eclipse.xtend.lib.macro.declaration.TypeReference
-import java.util.Map
-
-// TODO MK if active annotation for static delegation is added to Xtend then
-// merge all code of this class with org.eclipse.xtend.lib.annotations.DelegateProcessor$Util
```

Comment:
I’m sorry, it obviously was a little late yesterday. You are talking about [`@Delegate`](https://github.com/eclipse/xtext-lib/blob/master/org.eclipse.xtend.lib/src/org/eclipse/xtend/lib/annotations/Delegate.xtend), which was actually already released back in Xtend version 2.7. 

The `replace` methods in this PR were indeed copied from `DelegateProcessor` verbatim. However, `DelegateProcessor` is marked with `@noreference`, which 

> Indicates that clients must not reference this type (class, interface, enum, or annotation), method, constructor, or non-final field. Any code that directly invokes the associated method or constructor or references the associated non-final field will be flagged with a problem.

>When the tag is used on a type, it behaves as though the tag was added to any of the types members. For example, adding the tag to class will flag any references to any methods or non-final fields from that class. [[1]](https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.pde.doc.user%2Freference%2Fapi-tooling%2Fapi_javadoc_tags.htm)

So we *could* use the replace methods from `DelegateProcessor`, but that would break their contract. That’s why I think we should replace our `replace` methods with references to `DelegateProcessor`. I instead added a comment stating where they came from.

Is that alright?